app.rb

```diff
@@ -0,0 +1,47 @@
+require 'bundler'
+
+Bundler.require(:default)
+
+require 'sinatra/reloader'
+require 'fast_secure_compare/fast_secure_compare'
+
+enable :sessions
+
+get '/' do
+  if session[:loggedin]
+    redirect '/logged'
+  else
+    erb :index
+  end
+end
+
+post '/logged' do
+  line = File.read('password.txt')

```

Comment:
Jeśli nie przewidujemy zmiany hasła bez restartu serwera to ten plik mógłby być odczytywany z dysku tylko raz i wartość hasła przechowywana w stałej.